src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2781,21 +2978,47 @@ HRESULT CProfilerManager::InitializeForAttach(
 
     m_bAttach = true;
 
+    if (pvClientData == nullptr ||
+        cbClientData == 0)
+    {
+        // No configuration provided
+        return S_FALSE;
+    }
+
     vector<CComPtr<CConfigurationSource>> configSources;
 
-    // TODO: Parse configuration information from pvClientData.
+    // WCHAR = 2 bytes, hence we halve the byte count.
+    UINT charCount = cbClientData / 2;
+    UINT bufferSize = charCount + 1; // add room for null terminator.
+    unique_ptr<WCHAR[]> wszConfigXml = make_unique<WCHAR[]>(bufferSize);
+
+    LPCWSTR pData = reinterpret_cast<LPCWSTR>(pvClientData);
+    IfFailRetErrno(
+        memcpy_s(
+            wszConfigXml.get(), // Destination buffer
+            bufferSize * 2,     // Destination buffer size in Bytes
+            pData,              // Source buffer
+            cbClientData        // Source buffer size in Bytes
+        ));
+
+    m_wszConfigXml = wszConfigXml.release();
+    m_configSources = configSources;
```

Comment:
>m_configSources = configSources; [](start = 4, length = 32)

This shouldn't be necessary m_configSources should already be an empty vector. Copying from another empty vector should not be necessary.